src/main/java/frc/robot/poseestimator/limelights/VisionObservationFiltered.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.poseestimator.limelights;
+
+import frc.robot.constants.Field;
+import frc.robot.poseestimator.observations.VisionObservation;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VisionObservationFiltered extends GBSubsystem {
+
+	private LimelightRawData limelightHardware;
+	private VisionObservationFilteredConfig config;
+
+	public VisionObservationFiltered(VisionObservationFilteredConfig config) {
+		super(config.logPath());
+
+		this.limelightHardware = new LimelightRawData(config.limelightsNames(), config.hardwareLogPath());
+		this.config = config;
+	}
+
+	public List<VisionObservation> getFilteredVisionObservations() {
+		ArrayList<VisionObservation> estimates = new ArrayList<>();
+
+		for (LimelightData limelightData : limelightHardware.getAllAvailableLimelightData()) {
+			Logger.recordOutput(super.getLogPath() + limelightData.timeStamp(), limelightData.AprilTagHeight());
+			if (!filterOutLimelightData(limelightData)) {
+				double standardDeviation = getDynamicStandardDeviations(limelightData);
+				double[] standardDeviations = new double[] {standardDeviation};
```

Comment:
shouldn't the stdDevs have 3 values? insert the value 3 times

Reply:
I was told by @aaa-png to make it {standardDeviation, standardDeviation, standardDeviation}. That's what I changed it to, even though the theta probably should be the square of that.